applications/ShapeOptimizationApplication/python_scripts/response_functions/surface_normal_shape_change.py

```diff
@@ -0,0 +1,123 @@
+# ==============================================================================
+#  KratosShapeOptimizationApplication
+#
+#  License:         BSD License
+#                   license: ShapeOptimizationApplication/license.txt
+#
+#  Main authors:    Geiser Armin, https://github.com/armingeiser
+#
+# ==============================================================================
+
+import time as timer
+import KratosMultiphysics as KM
+import KratosMultiphysics.ShapeOptimizationApplication as KSO
+from KratosMultiphysics import Logger
+from KratosMultiphysics.response_functions.response_function_interface import ResponseFunctionInterface
+
+class SurfaceNormalShapeChange(ResponseFunctionInterface):
+    """
+    This simple geometric response functions calculates the sum of all shape updates in
+    surface normal direction. Shape update in surface normal direction is considered positive,
+    in the opposite direction negative.
+
+    Note that the direction of the surface normals changes in every step,
+    this is why the response value is not a meaningful measure but only an indication on how much the shape changed.
+
+    Important settings:
+    flip_normal_direction : boolean flag for changing on which side of the surface the update is considered positive.
+    """
+
+    def __init__(self, identifier, response_settings, model):
+        self.identifier = identifier
+
+        response_settings.ValidateAndAssignDefaults(self.GetDefaultSettings())
+
+        self.response_settings = response_settings
+        self.model = model
+
+        model_part_name = response_settings["model_part_name"].GetString()
+        input_type = response_settings["model_import_settings"]["input_type"].GetString()
+        if input_type == "mdpa":
+            self.model_part = self.model.CreateModelPart(model_part_name)
+            domain_size = response_settings["domain_size"].GetInt()
+            if domain_size not in [2, 3]:
+                raise Exception("SurfaceNormalShapeChange: Invalid 'domain_size': {}".format(domain_size))
+            self.model_part.ProcessInfo.SetValue(KM.DOMAIN_SIZE, domain_size)
+        elif input_type == "use_input_model_part":
+            self.model_part = self.model.GetModelPart(model_part_name)
+        else:
+            raise Exception("SurfaceNormalShapeChange: '{}' model part input type not implemented.".format(input_type))
```

Comment:
May be this can be refactored? I see this is being used few times in other py scripts.

Reply:
Thats true. I will create a new Issue/PR for this